### ISSUE TITLE
Webhook scheduled actions delayed

### ISSUE THREAD

**Issue by dstep2007** [woocommerce/woocommerce#22146]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1.  Update an order (or bulk update multiple orders) to the processing status.
2. Check the cron events list (using wp-crontrol plugin or wp cron events list on CLI) and you will see that the crons are not shown.

This issue has only come up since we upgraded to WC v3.5.2.  We were running a version prior to 3.5 before and it worked fine.  I've seen other issues related to the delete_version_transients cron that is now filling up my cron list.  Is it possible that because my cron list is getting overloaded with the delete_version_transients cron that it is affecting the webhook delivery crons?

## Expected/actual behavior

When I follow those steps, I see nothing in the wp cron events table for the woocommerce_deliver_webhook_async cron event that is usually created.

![image](https://user-images.githubusercontent.com/13855406/49544475-ee3d7780-f8a8-11e8-8462-c36b657e7f4b.png)

I was expecting to see the woocommerce_deliver_webhook_async be created

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

## WordPress Environment
<details>
WordPress 4.9.8
PHP 5.6
Flatsome Theme (child theme) v 3.6.1
```
`
### WordPress Environment ###

Home URL: https://healthyhempoil.com
Site URL: https://healthyhempoil.com
WC Version: 3.5.2
Log Directory Writable: ✔
WP Version: 4.9.8
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: –
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 5.6.38-3+ubuntu18.04.1+deb.sury.org+1 - We recommend using PHP version 7.2 or above for greater performance and security. How to update your PHP version
PHP Post Max Size: 100 MB
PHP Time Limit: 3600
PHP Max Input Vars: 1000
cURL Version: 7.58.0
OpenSSL/1.1.1

SUHOSIN Installed: –
MySQL Version: 5.6.41-84.1-log
Max Upload Size: 50 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.5.2
WC Database Prefix: wp_
Total Database Size: 2682.71MB
Database Data Size: 1394.69MB
Database Index Size: 1288.02MB
wp_woocommerce_sessions: Data: 38.06MB + Index: 2.50MB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_order_items: Data: 26.56MB + Index: 11.52MB
wp_woocommerce_order_itemmeta: Data: 167.69MB + Index: 112.33MB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.05MB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB
ts_redirects: Data: 0.02MB + Index: 0.00MB
wp_ac_guest_abandoned_cart_history: Data: 0.02MB + Index: 0.00MB
wp_affiliate_wp_affiliatemeta: Data: 0.02MB + Index: 0.03MB
wp_affiliate_wp_affiliates: Data: 0.02MB + Index: 0.02MB
wp_affiliate_wp_creatives: Data: 0.02MB + Index: 0.02MB
wp_affiliate_wp_referrals: Data: 0.02MB + Index: 0.02MB
wp_affiliate_wp_visits: Data: 0.02MB + Index: 0.02MB
wp_ak_404_log: Data: 0.02MB + Index: 0.00MB
wp_commentmeta: Data: 5.52MB + Index: 9.55MB
wp_comments: Data: 136.67MB + Index: 103.36MB
wp_ewwwio_images: Data: 0.09MB + Index: 0.09MB
wp_gf_draft_submissions: Data: 0.02MB + Index: 0.02MB
wp_gf_entry: Data: 2.52MB + Index: 0.39MB
wp_gf_entry_meta: Data: 5.52MB + Index: 5.58MB
wp_gf_entry_notes: Data: 0.02MB + Index: 0.03MB
wp_gf_form: Data: 0.02MB + Index: 0.00MB
wp_gf_form_meta: Data: 0.08MB + Index: 0.00MB
wp_gf_form_revisions: Data: 0.02MB + Index: 0.03MB
wp_gf_form_view: Data: 1.52MB + Index: 0.89MB
wp_headlines_variations: Data: 0.02MB + Index: 0.13MB
wp_itsec_lockouts: Data: 0.02MB + Index: 0.08MB
wp_itsec_log: Data: 2.02MB + Index: 0.45MB
wp_itsec_temp: Data: 0.02MB + Index: 0.06MB
wp_iwp_backup_status: Data: 0.02MB + Index: 0.00MB
wp_links: Data: 0.02MB + Index: 0.02MB
wp_nxs_log: Data: 0.05MB + Index: 0.00MB
wp_options: Data: 54.55MB + Index: 24.20MB
wp_pmxe_exports: Data: 3.52MB + Index: 0.00MB
wp_pmxe_google_cats: Data: 0.39MB + Index: 0.00MB
wp_pmxe_posts: Data: 15.55MB + Index: 0.00MB
wp_pmxe_templates: Data: 0.09MB + Index: 0.00MB
wp_pmxi_files: Data: 0.02MB + Index: 0.00MB
wp_pmxi_history: Data: 0.02MB + Index: 0.00MB
wp_pmxi_imports: Data: 0.02MB + Index: 0.00MB
wp_pmxi_posts: Data: 0.02MB + Index: 0.00MB
wp_pmxi_templates: Data: 0.02MB + Index: 0.00MB
wp_postmeta: Data: 368.95MB + Index: 419.00MB
wp_posts: Data: 51.56MB + Index: 25.09MB
wp_revslider_static_slides: Data: 0.02MB + Index: 0.00MB
wp_rg_aweber: Data: 0.02MB + Index: 0.02MB
wp_rg_form: Data: 0.02MB + Index: 0.00MB
wp_rg_form_meta: Data: 0.08MB + Index: 0.00MB
wp_rg_form_view: Data: 1.52MB + Index: 0.89MB
wp_rg_incomplete_submissions: Data: 0.02MB + Index: 0.02MB
wp_rg_lead: Data: 2.52MB + Index: 0.33MB
wp_rg_lead_detail: Data: 4.52MB + Index: 8.09MB
wp_rg_lead_detail_long: Data: 1.52MB + Index: 0.00MB
wp_rg_lead_meta: Data: 0.02MB + Index: 0.05MB
wp_rg_lead_notes: Data: 0.02MB + Index: 0.03MB
wp_snippets: Data: 0.02MB + Index: 0.00MB
wp_stream: Data: 113.66MB + Index: 114.39MB
wp_stream_meta: Data: 262.81MB + Index: 337.42MB
wp_termmeta: Data: 0.25MB + Index: 0.03MB
wp_terms: Data: 1.52MB + Index: 0.58MB
wp_term_relationships: Data: 1.41MB + Index: 1.50MB
wp_term_taxonomy: Data: 1.52MB + Index: 0.56MB
wp_usermeta: Data: 112.69MB + Index: 100.27MB
wp_users: Data: 5.52MB + Index: 7.55MB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB
wp_woocommerce_per_product_shipping_rules: Data: 0.02MB + Index: 0.00MB
wp_woopriceop_visitorlog: Data: 0.06MB + Index: 0.02MB
wp_woopriceop_visitorprices: Data: 0.02MB + Index: 0.02MB
wp_yoast_seo_links: Data: 1.47MB + Index: 0.23MB
wp_yoast_seo_meta: Data: 0.09MB + Index: 0.00MB
wp_yop2_pollmeta: Data: 0.02MB + Index: 0.02MB
wp_yop2_polls: Data: 0.02MB + Index: 0.00MB
wp_yop2_poll_answermeta: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_answers: Data: 0.02MB + Index: 0.03MB
wp_yop2_poll_bans: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_custom_fields: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_logs: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_questionmeta: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_questions: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_results: Data: 0.02MB + Index: 0.02MB
wp_yop2_poll_templates: Data: 1.52MB + Index: 0.00MB
wp_yop2_poll_votes_custom_fields: Data: 0.02MB + Index: 0.02MB

### Post Type Counts ###

acf: 1
attachment: 2779
blocked_user_lite: 9
blocks: 4
customize_changeset: 20
custom_css: 3
dwqa-answer: 2
dwqa-question: 3
follow_up_email: 2
gfw_report: 3
landing_page: 1
nav_menu_item: 135
ni-order-status: 3
oembed_cache: 9
page: 109
popup: 1
popup_theme: 6
post: 327
product: 157
product_variation: 468
revision: 20
scheduled-action: 5351
shop_coupon: 347
shop_order: 116105
shop_order_refund: 1790
smart_offers: 1
subscription: 397
sub_transaction: 1692
wc_order_email: 2
wc_order_status: 14
wc_zapier_feed: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (46) ###

ManageWP - Worker: by ManageWP – 4.6.3
Gravity Forms: by rocketgenius – 2.4.1
404page - your smart custom 404 error page: by Peter Raschendorfer – 7
Revenue Conduit: by Revenue Conduit – 1.0.0 – Not tested with the active version of WooCommerce
ActiveCampaign: by ActiveCampaign – 7.0
Ad Blocker Notify Lite: by Admiral – 2.2.3
Advanced noCaptcha & invisible Captcha: by Shamim Hasan – 2.8
Better Font Awesome: by Mickey Kay – 1.7.1
CartHook for WooCommerce: by CartHook – 1.0.3 – Not tested with the active version of WooCommerce
Code Snippets: by Shea Bunge – 2.12.1
Coming Soon Page & Maintenance Mode by SeedProd: by SeedProd – 5.0.23
WooCommerce Coupon Generator: by Jeroen Sormani – 1.0.1 – Not tested with the active version of WooCommerce
Custom Related Products for WooCommerce: by Scott Nelle – 1.3 – Not tested with the active version of WooCommerce
Disable Cart Fragments: by LittleBizzy – 1.3.0
WooCommerce Discontinued Products: by Owl Computing – 1.0.0 – Not tested with the active version of WooCommerce
Disqus for WordPress: by Disqus – 3.0.16
Fast Velocity Minify: by Raul Peixoto – 2.4.5
MonsterInsights - eCommerce Addon: by MonsterInsights Team – 7.0.12
MonsterInsights Pro: by MonsterInsights – 7.3.2
Healthy Hemp Oil - WooCommerce Custom Shipping Countries: by Leho Kraav – 2017.02.24 – Not tested with the active version of WooCommerce
Social Login Buttons by Heateor: by Team Heateor – 1.1
iThemes Security Pro: by iThemes – 3.7.4
Collapse-O-Matic: by twinpictures
baden03 – 1.7.9

Max Mega Menu: by megamenu.com – 2.5.2
Rich Text Tags, Categories, and Taxonomies: by Katz Web Services
Inc. – 1.8

SumoMe: by SumoMe – 1.30
Super Socializer: by Team Heateor – 7.12.5
Tawk.to Live Chat: by Tawkto – 0.3.4
Compress JPEG & PNG images: by TinyPNG – 3.1.0
User Role Editor: by Vladimir Garagulya – 4.47
Woocommerce Blocker Lite: by multidots – 1.6.8 – Not tested with the active version of WooCommerce
WOO New: by WP Fix It – 1.0
WooCommerce Address Validation: by SkyVerge – 2.3.1 – 2.3.3 is available – Not tested with the active version of WooCommerce
WooCommerce Conditional Shipping and Payments: by SomewhereWarm – 1.3.5 – 1.5.0 is available – Not tested with the active version of WooCommerce
WooCommerce Conversion Tracking: by Tareq Hasan – 2.0 – Not tested with the active version of WooCommerce
WooCommerce Offline Credit Card Processing: by WP Lab – 1.7.2 – Not tested with the active version of WooCommerce
WooCommerce Hide Password Protected Products: by Barry Kooij - WooThemes – 1.0 – Not tested with the active version of WooCommerce
WooCommerce Order Status Manager: by SkyVerge – 1.9.2 – 1.9.3 is available – Not tested with the active version of WooCommerce
WooCommerce Shipping Per Product v2: by WooCommerce – 2.2.13 – 2.2.15 is available – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.5.2
NMI Gateway for WooCommerce: by PatSaTECH – 1.7.0 – Not tested with the active version of WooCommerce
WordPress Importer: by wordpressdotorg – 0.6.4
Yoast SEO Premium: by Team Yoast – 9.2.1
WP Crontrol: by John Blackbourn & contributors – 1.6.2
WP Htaccess Editor: by Lukenzi – 1.3.0
Yotpo Social Reviews for Woocommerce: by Yotpo – 1.1.8 – Not tested with the active version of WooCommerce

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #98 - /shop/
Cart: #90835 - /cart/
Checkout: #90834 - /checkout/
My account: #90833 - /my-account/
Terms and conditions: #90847 - /terms-conditions/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.6.1
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php version 3.1.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/cart/cart.php version 3.4.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/checkout/form-checkout.php version 3.0.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/checkout/form-coupon.php version 3.4.0 is out of date. The core version is 3.4.4
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
flatsome-child/woocommerce/emails/customer-completed-order.php version 2.5.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome-child/woocommerce/myaccount/form-login.php version 3.2.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php version 3.3.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/notices/notice.php version 1.6.4 is out of date. The core version is 3.5.0
flatsome/woocommerce/notices/success.php version 3.3.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php version 3.3.2 is out of date. The core version is 3.5.1
flatsome/woocommerce/single-product/product-thumbnails.php version 3.3.2 is out of date. The core version is 3.5.1
flatsome/woocommerce/single-product/rating.php
flatsome-child/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php version 1.6.4 is out of date. The core version is 3.5.0
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php version 3.2.0 is out of date. The core version is 3.5.0
flatsome/woocommerce/single-product.php

Outdated Templates: ❌
					
					
						Learn how to update
`
```
</details>

**Comment by achadee** [comment]
We are also having this issue with customers that have updated to 3.5.x, downgrading back does not fix the issue. No webhook log file is being created.

**Comment by dstep2007** [comment]
Just an update, I just found this new (to me) section on the Woocommerce -> Status -> Scheduled Actions page.  I have 670 pending webhook deliveries waiting to be sent.  It looks like they all keep getting rescheduled?

![image](https://user-images.githubusercontent.com/13855406/49590822-d31b4800-f93a-11e8-951f-91949c7a37ba.png)

This brings up a few questions:
1. Are webhooks not using CRON anymore?
2. Why do they keep getting rescheduled? This is a HIGH volume store, is that causing things to get backed up?
3. Is there something server configuration wise I can do to allow these webhooks to fire better with this new setup?
4. Where can I find information on this new webhook setup so I can dig into it further and see if I can figure out why these aren't firing?

Thanks!

**Comment by dstep2007** [comment]
So I dug a even further into this.  Woocommerce is using Action Scheduler now as it's task runner for Webhooks.  What I would like to know is if this plugin (by the creators of Action Scheduler) is suitable for use with WooCommerce?

https://github.com/prospress/action-scheduler-high-volume

It is designed to increase performance on high volume sites.  But has it been tested with WooCommerce??

**Comment by xcelerated1** [comment]
I'm having the same issues on a high volume membership/subscription site.

**Comment by kloon** [comment]
> So I dug a even further into this. Woocommerce is using Action Scheduler now as it's task runner for Webhooks. What I would like to know is if this plugin (by the creators of Action Scheduler) is suitable for use with WooCommerce?

It definitely has been tested, it is originally developed by the WooCommerce subscription developers and it is used to processes subscription renewals reliably. Other WooCommerce extensions also use it to process tasks async.

If you have a large volume site you can follow the guide at https://github.com/prospress/action-scheduler-high-volume which shows you how to increase the number of tasks to process at a time which should speed things up. Event with the default threshold the webhooks will eventually be delivered once the schedule works through the backlog of tasks.

**Comment by dstep2007** [comment]
@kloon I'm not sure this issue is at a point where it can be closed...

Even with the high volume plugin added to my site, and adjustments made for my hosting (WP Engine), I'm still seeing backlogs in the thousands that are taking over a day to complete.  What else can I do to get this working properly? What other information can I provide to you to help?  These webhooks are critical to my websites operation and fulfillment process.

**Comment by dstep2007** [comment]
@kloon should I make a new issue regarding the scheduled actions not running?  Maybe that's why you listed this as closed?  I'd be glad to make a new issue if you'd like.

**Comment by kloon** [comment]
@dstep2007 have you followed the steps as per https://actionscheduler.org/perf/ I would say try increasing the batch size and concurrent batches even more and see if things catch up for you. If you have higher end hosting you should be able to push those numbers up quite high to keep up with the high traffic volume on your site.

**Comment by xcelerated1** [comment]
I'm having the very same issues as @dstep2007 and we have also implemented the same recommended settings. Same issue. Backlog and the site runs super slow.

I'm also on WP Engine, P5 level enterprise dedicated server.

**Comment by dstep2007** [comment]
@kloon yes I've tried the steps at that link.  No luck so far.  When I first installed the plugin and updated the settings for WP Engine, it appeared to be helping, but then it just kept getting worse and worse.

I've also tried using the WP CLI commands with the "--force" flag to ignore any server restrictions based on memory/resource usage as well as batch size increases, and that doesn't even push the actions through faster.

I also noticed that the site had multiple webhooks running (about 15 or so) based on other topics (order created, product updated, etc.) so I disabled those thinking less hooks would make my critical webhooks run better.  It did not help.

@xcelerated1 I am on a very robust shared server with my site.  I feared that resource usage was my issue, but like a stated above, even using the --force flag which ignores resource usage did not push my hooks through faster.  I'll keep this thread posted if I find something that works.

**Comment by kloon** [comment]
@thenbrent just tagging you in on this for some feedback if possible. Seems like it is mostly WPEngine customers experiencing the issue.

**Comment by ellisar** [comment]
@kloon we're running on digital ocean (16 GB Memory / 320 GB Disk - Ubuntu 16.04.2 x64 ) and seeing the exact issue with our subscription site. Currently have 9,024 scheduled actions that are delayed / past the scheduled date (some still dating back to 2018-12-03 17:46:40 +00:00 (1 week 5 days ago). We've thrown a lot of resources to debugging it but still no luck. 

We had to manually renew about half of our subscriptions via Scheduled Actions page with the help of a macro as only about 200 renewals were processing each day (all originally scheduled for Dec 1st) – it was pretty painful. We will test https://github.com/prospress/action-scheduler-high-volume but it seems like it is not a complete fix based on the testing of @xcelerated1 and @dstep2007.

We turned off all non-critical webhooks to try to minimize the delayed queue but still pretty far in the hole. Does anyone know a way to bulk delete certain non-critical scheduled actions to help speed the catch-up process?

**Comment by ellisar** [comment]
@kloon @dstep2007 @xcelerated1 Success!! The Action Scheduler High Volume install fixed it for us on Digital Ocean – thank god. Thank you all for your help / documentation of this issue – that was a rough few weeks. It definitely kicked the server into high gear, using 80% of the CPU now. I installed the plugin about 90 minutes ago and it's already processed 1,600 of the missed actions. Before installing, it was only processing ~30/hr of the backlogged actions and now it's processing over 1000/hr.
<img width="817" alt="screenshot 2018-12-16 07 40 25" src="https://user-images.githubusercontent.com/5906102/50055752-de732e00-0107-11e9-98d0-8466110b7d0b.png">

**Comment by dstep2007** [comment]
Just to give an update on my findings, I've temporarily disabled about 90% of my non-critical plugins.  Down from about 20 webhooks on Woocommerce topics to 2 of them.  I am up from processing about only 10/hr when I had all hooks enabled to running very successfully 70/hr.

Now that my actions backlog is also completely clear and up to speed, I'm going to attempt re-introducing the other webhooks one by one to see if it can continue to handle the increased amount of actions.  Ideally, I can run 1000's/hr like the plugin says it's capable of doing and like @ellisar shows it's been doing for him.  I'll report back on my results.

**Comment by mikejolley** [comment]
This is a related bug/issue/patch https://github.com/woocommerce/woocommerce/pull/21427 For next update.

**Comment by thenbrent** [comment]
@dstep2007 (and others coming to this) I believe your core issue was being caused by the issue outlined in #22087. This has a fixed already merged in #22088 (which I believe will go out in the next version).

It sounds like there two related but separate issues being mixed together here:

1. a large queue
1. the queue being processed slowly

Both contribute to the same symptom - webhooks being processed slowly. But they are separate issues.

The first issue of overly large queues is fixed with #22088. The 2nd part is helped with the Action Scheduler High Volume plugin and other [performance tweaking](https://actionscheduler.org/perf/), but it's expected that is only required on large stores (i.e. order volume of thousands per month, not dozens or hundreds).

> I am up from processing about only 10/hr when I had all hooks enabled to running very successfully 70/hr.

If your site is only able to process 70 / hour, and it's not expected that it will take a really long time to process each webhook (over a minute), then something is definitely up that will need further investigation. Especially if that processing rate is so slow even when using WP CLI.

Please open an issue on http://github.com/Prospress/action-scheduler/ and we can investigate.
